[change] Update README demos and add example GIFs (#441)
openwisp/netjsongraph.js: README.md

[![Coverage Status](https://coveralls.io/repos/github/openwisp/netjsongraph.js/badge.svg?branch=master)](https://coveralls.io/github/openwisp/netjsongraph.js?branch=master)
![Language](https://img.shields.io/badge/language-javascript-orange.svg)

Leverage the power of [EchartsJS](https://github.com/apache/incubator-echarts) and [LeafletJS](https://github.com/Leaflet/Leaflet) to visualize network topology using the
[NetJSON](http://netjson.org) `NetworkGraph` format.

Build powerful and interoperable visualizations without losing flexibility!

> ## Example Demos

Reviewer: ```suggestion
## Live Demos
```

> Explore the demos below. Click a preview to open.

Reviewer: ```suggestion
Find below a rich set of live demos for the main use cases supported by NetJSONGraph.js.
```

> ### Basic usage

Reviewer: Let's make the title reflect what's written in the description:

```suggestion
### Network graph
```

[![Basic usage](docs/gifs/netjsongraph.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsongraph.html)

Visualize a NetJSON NetworkGraph with pan/zoom, labels and tooltips.

### Geographic map

[![Geographic map](docs/gifs/netjsonmap.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsonmap.html)

Plot nodes by geographic coordinates on a Leaflet basemap; pan/zoom with clustered markers.

### Multiple interfaces

[![Multiple interfaces](docs/gifs/netjson-multipleInterfaces.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjson-multipleInterfaces.html)

Deduplicate, merge interfaces and enrich link info before rendering.

### Search elements

[![Search elements](docs/gifs/netjson-searchElements.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjson-searchElements.html)

Fetch filtered data and update the view in-place.

### Date parse

[![Date parse](docs/gifs/netjson-dateParse.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjson-dateParse.html)

> Normalize ISO timestamps to the browser timezone in node/link details.

Reviewer: Can we merge this example with the network graph example and get rid of this?

> ### Switch render mode

Reviewer: ```suggestion
### Switch render mode (Canvas/SVG)
```

[![Switch render mode](docs/gifs/netjson-switchRenderMode.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjson-switchRenderMode.html)

Toggle the renderer to balance quality/performance.

> ### Switch graph mode

Reviewer: ```suggestion
### Switch graph mode (Network Graph / Geographic Map)
```

[![Switch graph mode](docs/gifs/netjson-switchGraphMode.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjson-switchGraphMode.html)

Switch between force-directed graph and geographic map views.

### Nodes expand or fold

[![Nodes expand or fold](docs/gifs/netjsongraph-nodeExpand.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsongraph-nodeExpand.html)

Expand neighbors on click to focus on subgraphs; fold to declutter.

> ### Indoor map

Reviewer: Let's move this example up (eg: after geographic map).

Can you find a different floorplan image here? Maybe @dee077 can suggest a nice one to use?

[![Indoor map](docs/gifs/netjsonmap-indoormap.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsonmap-indoormap.html)

> Use a 2D floorplan (Leaflet CRS.Simple) to place and connect indoor nodes.

Reviewer: ```suggestion

This example demonstrates the visualization of a mesh network topology graph on a 2D floorplan image.
```

### Leaflet plugins

[![Leaflet plugins](docs/gifs/netjsonmap-plugins.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsonmap-plugins.html)

Use draw/measure and other Leaflet plugins in map mode.

> ### Custom attributes

Reviewer: ```suggestion
### Network Graph with custom attributes and legend
```

[![Custom attributes](docs/gifs/netjsongraph-elementsLegend.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsongraph-elementsLegend.html)

> Style nodes/links by categories and present them in a legend.

Reviewer: ```suggestion
This example shows how to use multiple nodes/links categories and explain these in a legend.
```

[![JSONDataUpdate using override option](docs/gifs/netjsonmap-nodeTiles.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsonmap-nodeTiles.html)

Update features/styles by zoom using JSONDataUpdate.

### JSONDataUpdate using append option

[![JSONDataUpdate using append option](docs/gifs/netjsonmap-appendData.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsonmap-appendData.html)

Append new nodes/links on the fly without a full reload.

### Multiple tiles render

[![Multiple tiles render](docs/gifs/netjsonmap-multipleTiles.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsonmap-multipleTiles.html)

Switch between different tile providers and styles.

### Append data using arrays

[![Append data using arrays](docs/gifs/netjsonmap-appendData2.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsonmap-appendData2.html)

Append data from an array of sources at startup.

### Geographic map with GeoJSON data

[![Geographic map with GeoJSON data](docs/gifs/njg-geojson.gif)](https://openwisp.github.io/netjsongraph.js/examples/njg-geojson.html)

Load GeoJSON FeatureCollections; they are converted to NetJSON nodes/links automatically.

> ### Clustering

Reviewer: Geographic Map Clustering

[![Clustering](docs/gifs/netjson-clustering.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjson-clustering.html)

> Aggregate nearby nodes into clusters; automatic overlap prevention for identical coordinates.

Reviewer: ```suggestion
Aggregate nearby nodes into clusters with automatic overlap prevention of clusters having different colors.
```

### WiFi Clients Graph

[![WiFi Clients Graph](docs/gifs/netjsongraph-wifi-clients.gif)](https://openwisp.github.io/netjsongraph.js/examples/netjsongraph-wifi-clients.html)

> Show per-node client counts as small colored dots around each node.

Reviewer: ```suggestion
This example demonstrates how to visualize per-node client count as small colored dots around each node.

This can be useful for showing WiFi Clients of mesh nodes, or Layer 2 Neighbors of TAP VPN clients. 
```
